### ISSUE TITLE
Block-wise transfer not complete when a different request is sent during the transfer

### ISSUE THREAD

**Issue by jqiaobln** [obgm/libcoap#770]
## Environment

- libcoap version (run ``git describe --tags`` to find it):
v4.3.0-57-gdf9071c
- Build System:             Make
- Operating System:         Linux
- Operating System Version: 20.04
- Hosted Environment:       None

## Problem Description
A post request and a block-wise transfer for uploading a file of 4KB size were tested separately and they both completed successfully. However if I started the post request immediately after the block-wise transfer has begun but not yet finished, the post request still completed successfully however the block-wise transfer didn't continue after the server responded with ACK. I have tried with NSTART set to 1 and 2, the behaviours are the same. 

### Expected Behavior
The block-wise transfer should not pause. It should continue transferring the blocks until file uploading is completed. When NSTART=1, the post request should not begin until all blocks are transmitted as there can be only 1 transaction in flight. When NSTART = 2, the post request is sent immediately. 

### Actual Behavior
The post request still completed successfully. However block-wise transfer always paused after the 2nd block is sent and ACK is received.

### Steps to reproduce

1. Prepare a libcoap client which stars a block-wise transfer and sends a post request within 1 DTLS session. 
2. Run the client

### Code to reproduce this issue
See attachment.
[blocktransfer.zip](https://github.com/obgm/libcoap/files/7517718/blocktransfer.zip)

## Debug Logs
Complete logs in attachment.

Nov 10 15:03:00.541 DEBG *  192.168.0.33:50465 <-> 54.234.173.90:5784 DTLS: received 49 bytes
v:1 t:ACK c:2.31 i:ab8e {0200000001} [ Block1:2/M/512 ]
Nov 10 15:03:00.541 DEBG ** 192.168.0.33:50465 <-> 54.234.173.90:5784 DTLS: mid=0xab8e: removed
Nov 10 15:03:00.541 DEBG ** process incoming 2.31 response:
Nov 10 15:03:01.127 DEBG ** 192.168.0.33:50465 <-> 54.234.173.90:5784 DTLS: lg_crcv 0x5583792fc390 released
Nov 10 15:04:30.122 INFO timeout
Nov 10 15:04:30.123 DEBG *  192.168.0.33:50465 <-> 54.234.173.90:5784 DTLS: sent 39 bytes
Nov 10 15:04:30.123 DEBG ***EVENT: 0x0000
Nov 10 15:04:30.123 DEBG ***192.168.0.33:50465 <-> 54.234.17
3.90:5784 DTLS: session closed

## Other items if possible

- The only configuration I changed is NSTART.
- Network configuration doesn't seem to be an issue.

**Comment by mrdeep1** [comment]
Thans for providing all the information needed to reproduce the issue.  I am looking into why lg_xmit structure is prematurely released.

**Comment by mrdeep1** [comment]
The issue is fixed in PR #740 which is waiting to be merged into the develop branch.

**Comment by jqiaobln** [comment]
> The issue is fixed in PR #740 which is waiting to be merged into the develop branch.

Great news. Thanks for the response.

I do have 1 question regarding how 2 transactions are handled concurrently: when NSTART=1, why is the post request sent between block transfers? As there should be only 1 transaction in flight which is block wise transfer, as block transfer can only be done sequentially.

**Comment by jqiaobln** [comment]
Thanks for the reply. Is there an estimated time this PR will get merged?

**Comment by obgm** [comment]
PR #740 now has been merged. Can you confirm that this issue can be closed off?

**Comment by jqiaobln** [comment]
> PR #740 now has been merged. Can you confirm that this issue can be closed off?

Will give it a try and get back to you.

**Comment by jqiaobln** [comment]
I seem to encounter a new issue with the current master: block-wise transfer no more working.
Options used:
operation: post
content type: application/octet-stream

==============================================================
Detail log output:
15:07Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2755: 0x555f67a2ee30: => flush output
Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2773: 0x555f67a2ee30: message length: 1157, out_left: 1157
Dec 17 15:04:01.219 DEBG * 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: sent 1157 bytes
Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2779: 0x555f67a2ee30: ssl->f_send() returned 1157 (-0xfffffb7b)
Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2807: 0x555f67a2ee30: <= flush output
Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3476: 0x555f67a2ee30: <= write record
Dec 17 15:04:01.219 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:8710: 0x555f67a2ee30: <= write
**v:1 t:CON c:POST i:1240 {01000000000003} [ Uri-Host:coap-lb-na.play.test.cc, Uri-Port:5784, Uri-Path:msg, Uri-Path:v1, Uri-Path:did, Uri-Path:123456789, Uri-Path:deviceContext, Content-Format:application/octet-stream, Block1:0/M/1024, Size1:4692 ] :: binary data length 1024**
<<416c6c27732057656c6c205468617420456e64732057656c6c0a0a7072696e742f73617665207072696e742f7361766520766965770a2020092020095e20506c6179206d656e7520094e657874207363656e65205d2009200a2d2d2d0a202009202020202020090a0a41637420492c205363656e6520310a0a526f7573696c6c6f6e2e2054686520434f554e54e28099732070616c6163652e0a0920202020202009200a2d2d2d0a0a456e746572204245525452414d2c2074686520434f554e54455353206f6620526f7573696c6c6f6e2c2048454c454e412c5d205b705d616e64204c414645552c20616c6c20696e20626c61636b5d0a0a20202020436f756e746573732e20496e2064656c69766572696e67206d7920736f6e2066726f6d206d652c204920627572792061207365636f6e642068757362616e642e200a0a202020204265727472616d2e20416e64204920696e20676f696e672c206d6164616d2c2077656570206f276572206d792066617468657227732064656174680a20202020616e65773a206275742049206d75737420617474656e6420686973206d616a65737479277320636f6d6d616e642c20746f20350a2020202077686f6d204920616d206e6f7720696e20776172642c20657665726d6f726520696e207375626a656374696f6e2e200a0a202020204c616665752e20596f75207368616c6c2066696e64206f6620746865206b696e6720612068757362616e642c206d6164616d3b20796f752c0a202020207369722c2061206661746865723a206865207468617420736f2067656e6572616c6c7920697320617420616c6c2074696d65730a20202020676f6f64206d757374206f66206e656365737369747920686f6c64206869732076697274756520746f20796f753b2077686f73650a20202020776f727468696e65737320776f756c6420737469722069742075702077686572652069742077616e746564207261746865722031300a202020207468616e206c61636b2069742077686572652074686572652069732073756368206162756e64616e63652e200a0a20202020436f756e746573732e205768617420686f7065206973207468657265206f6620686973206d616a65737479277320616d656e646d656e743f200a0a202020204c616665752e2048652068617468206162616e646f6e6564206869732070687973696369616e732c206d6164616d3b20756e6465722077686f73650a20202020707261637469736573206865206861746820706572736563757465642074696d65207769746820686f70652c20616e640a2020202066696e6473206e6f206f7468657220616476616e7461676520696e207468652070726f6365737320627574206f6e6c79207468652031350a202020206c6f73696e>>
<<A l l ' s W e l l T h a t E n d s W e l l . . p r i n t / s a v e p r i n t / s a v e v i e w . . . ^ P l a y m e n u . N e x t s c e n e ] . . - - - . . . . . A c t I , S c e n e 1 . . R o u s i l l o n . T h e C O U N T . . . s p a l a c e . . . . . - - - . . E n t e r B E R T R A M , t h e C O U N T E S S o f R o u s i l l o n , H E L E N A , ] [ p ] a n d L A F E U , a l l i n b l a c k ] . . C o u n t e s s . I n d e l i v e r i n g m y s o n f r o m m e , I b u r y a s e c o n d h u s b a n d . . . B e r t r a m . A n d I i n g o i n g , m a d a m , w e e p o ' e r m y f a t h e r ' s d e a t h . a n e w : b u t I m u s t a t t e n d h i s m a j e s t y ' s c o m m a n d , t o 5 . w h o m I a m n o w i n w a r d , e v e r m o r e i n s u b j e c t i o n . . . L a f e u . Y o u s h a l l f i n d o f t h e k i n g a h u s b a n d , m a d a m ; y o u , . s i r , a f a t h e r : h e t h a t s o g e n e r a l l y i s a t a l l t i m e s . g o o d m u s t o f n e c e s s i t y h o l d h i s v i r t u e t o y o u ; w h o s e . w o r t h i n e s s w o u l d s t i r i t u p w h e r e i t w a n t e d r a t h e r 1 0 . t h a n l a c k i t w h e r e t h e r e i s s u c h a b u n d a n c e . . . C o u n t e s s . W h a t h o p e i s t h e r e o f h i s m a j e s t y ' s a m e n d m e n t ? . . L a f e u . H e h a t h a b a n d o n e d h i s p h y s i c i a n s , m a d a m ; u n d e r w h o s e . p r a c t i s e s h e h a t h p e r s e c u t e d t i m e w i t h h o p e , a n d . f i n d s n o o t h e r a d v a n t a g e i n t h e p r o c e s s b u t o n l y t h e 1 5 . l o s i n >>
Dec 17 15:04:01.221 DEBG ** 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: mid=0x1240: added to retransmit queue (2250ms)
Dec 17 15:04:01.290 DEBG * 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: received 51 bytes
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:8270: 0x555f67a2ee30: => read
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2755: 0x555f67a2ee30: => flush output
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2767: 0x555f67a2ee30: <= flush output
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:0081: 0x555f67a2ee30: set_timer to 0 ms
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4311: 0x555f67a2ee30: => read record
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2536: 0x555f67a2ee30: => fetch input
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2596: 0x555f67a2ee30: in_left: 0, nb_want: 13
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2638: 0x555f67a2ee30: f_recv_timeout: 0 ms
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2646: 0x555f67a2ee30: ssl->f_recv(_timeout)() returned 51 (-0xffffffcd)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2742: 0x555f67a2ee30: <= fetch input
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4047: 0x555f67a2ee30: dumping 'input record header' (13 bytes)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4047: 0x555f67a2ee30: 0000: 17 fe fd 00 01 00 00 00 00 00 01 00 26 ............&
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4053: 0x555f67a2ee30: input record: msgtype = 23, version = [3:3], msglen = 38
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2536: 0x555f67a2ee30: => fetch input
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2596: 0x555f67a2ee30: in_left: 51, nb_want: 51
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2604: 0x555f67a2ee30: <= fetch input
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4232: 0x555f67a2ee30: dumping 'input record from network' (51 bytes)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4232: 0x555f67a2ee30: 0000: 17 fe fd 00 01 00 00 00 00 00 01 00 26 00 01 00 ............&...
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4232: 0x555f67a2ee30: 0010: 00 00 00 00 01 f2 b9 42 14 40 d9 8c 29 09 a5 fa .......B.@..)...
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4232: 0x555f67a2ee30: 0020: e7 e1 32 03 fb ff 23 1f 5d 3b e3 de 23 dd 6f 89 ..2...#.];..#.o.
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4232: 0x555f67a2ee30: 0030: 09 2b b9 .+.
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1794: 0x555f67a2ee30: => decrypt buf
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1881: 0x555f67a2ee30: dumping 'additional data for AEAD' (13 bytes)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1881: 0x555f67a2ee30: 0000: 00 01 00 00 00 00 00 01 17 fe fd 00 0e .............
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1910: 0x555f67a2ee30: dumping 'IV used' (12 bytes)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1910: 0x555f67a2ee30: 0000: 3c ad 00 fa 00 01 00 00 00 00 00 01 <...........
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1911: 0x555f67a2ee30: dumping 'TAG used' (16 bytes)
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1911: 0x555f67a2ee30: 0000: 03 fb ff 23 1f 5d 3b e3 de 23 dd 6f 89 09 2b b9 ...#.];..#.o..+.
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2376: 0x555f67a2ee30: <= decrypt buf
Dec 17 15:04:01.290 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4259: 0x555f67a2ee30: dumping 'input payload after decrypt' (14 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4259: 0x555f67a2ee30: 0000: 67 5f 12 40 01 00 00 00 00 00 03 d1 0e 0e g_.@..........
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:4385: 0x555f67a2ee30: <= read record
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:0081: 0x555f67a2ee30: set_timer to 0 ms
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:8558: 0x555f67a2ee30: <= read
**v:1 t:ACK c:2.31 i:1240 {01000000000003} [ Block1:0/M/1024 ]**
Dec 17 15:04:01.291 DEBG ** 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: mid=0x1240: removed
**Dec 17 15:04:01.291 DEBG found Block option, block size is 1024, block nr. 0
Dec 17 15:04:01.291 WARN coap_pdu_resize: pdu too big**
Dec 17 15:04:01.291 DEBG ** 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: lg_xmit 0x555f67a2e4a0 released
Dec 17 15:04:01.291 DEBG ** 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: lg_crcv 0x555f67a2e610 released
Dec 17 15:04:01.291 DEBG **** process incoming 2.31 response:**
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:8725: 0x555f67a2ee30: => write close notify
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:5250: 0x555f67a2ee30: => send alert message
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:5251: 0x555f67a2ee30: send alert level=1 message=0
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3343: 0x555f67a2ee30: => write record
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1445: 0x555f67a2ee30: => encrypt buf
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1455: 0x555f67a2ee30: dumping 'before encrypt: output payload' (2 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1455: 0x555f67a2ee30: 0000: 01 00 ..
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1574: 0x555f67a2ee30: dumping 'additional data for AEAD' (13 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1574: 0x555f67a2ee30: 0000: 00 01 00 00 00 00 00 02 15 fe fd 00 02 .............
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1604: 0x555f67a2ee30: dumping 'IV used (internal)' (12 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1604: 0x555f67a2ee30: 0000: a7 e3 2a 43 00 01 00 00 00 00 00 02 ..*C........
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1606: 0x555f67a2ee30: dumping 'IV used (transmitted)' (8 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1606: 0x555f67a2ee30: 0000: 00 01 00 00 00 00 00 02 ........
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1616: 0x555f67a2ee30: before encrypt: msglen = 10, including 0 bytes of padding
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1643: 0x555f67a2ee30: dumping 'after encrypt: tag' (16 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1643: 0x555f67a2ee30: 0000: e9 e1 82 15 a6 59 d5 29 45 32 9f 14 cd fe e6 58 .....Y.)E2.....X
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:1781: 0x555f67a2ee30: <= encrypt buf
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3420: 0x555f67a2ee30: output record: msgtype = 21, version = [254:253], msglen = 26
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3425: 0x555f67a2ee30: dumping 'output record sent to network' (39 bytes)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3425: 0x555f67a2ee30: 0000: 15 fe fd 00 01 00 00 00 00 00 02 00 1a 00 01 00 ................
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3425: 0x555f67a2ee30: 0010: 00 00 00 00 02 40 bc e9 e1 82 15 a6 59 d5 29 45 .....@......Y.)E
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3425: 0x555f67a2ee30: 0020: 32 9f 14 cd fe e6 58 2.....X
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2755: 0x555f67a2ee30: => flush output
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2773: 0x555f67a2ee30: message length: 39, out_left: 39
Dec 17 15:04:01.291 DEBG * 192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: sent 39 bytes
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2779: 0x555f67a2ee30: ssl->f_send() returned 39 (-0xffffffd9)
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:2807: 0x555f67a2ee30: <= flush output
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:3476: 0x555f67a2ee30: <= write record
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:5263: 0x555f67a2ee30: <= send alert message
Dec 17 15:04:01.291 CIPH /build/mbedtls-mygkMM/mbedtls-2.16.4/library/ssl_tls.c:8741: 0x555f67a2ee30: <= write close notify
Dec 17 15:04:01.291 DEBG ***EVENT: 0x0000
Dec 17 15:04:01.291 DEBG ***192.168.0.196:50786 <-> 52.201.245.181:5784 DTLS: session closed

**Comment by mrdeep1** [comment]
I assume you need the latest develop code.

> Dec 17 15:04:01.291 WARN coap_pdu_resize: pdu too big

Is the clue here - it looks like you have found a corner case where the logic is trying to add in something which is too large into the PDU.

I will try to reproduce what you are seeing.  Can I get access to coap-lb-na.play.test.cc?
For debugging this, you only need to run with the -v7 (not -v9) option as it is not an MbedTLS issue.

**Comment by mrdeep1** [comment]
I have found an corner case issue which possibly can explain seeing `WARN coap_pdu_resize: pdu too big`, but I was unable to reproduce what you were seeing.  If you could try out Fix #796 to see if this clears your issue, that would be great and please let us know.

**Comment by mrdeep1** [comment]
@jqiaobln Now that #796 is in the develop branch, are you still having issues?

**Comment by jqiaobln** [comment]
> @jqiaobln Now that #796 is in the develop branch, are you still having issues?

I don't have access to the server currently. Will see what I can do.

**Comment by mrdeep1** [comment]
@jqiaobln Any updates on this?

**Comment by mrdeep1** [comment]
@jqiaobln Please free to re-open this  issue if there still is a problem.
